bin/simple_engine.ml

```diff
@@ -16,31 +16,32 @@ let runtime_counter device tones _domain_id ts counter _value =
   match counter with
   | EV_C_MINOR_PROMOTED ->
       starting_time := Some ts;
-      Midi.(write_output device [ message_on ~note:(tones 0) ~timestamp:0l () ])
+      Midi.(
+        write_output device
+          [
+            message_on ~note:(tones 0).Scale.note ~timestamp:0l
+              ~volume:(tones 0).Scale.volume ();
+          ])
       (* Unix.sleep 5;
          Midi.(write_output [ message_off ~note:base_note () ]) *)
   | _ -> ()
 
 let runtime_begin device tones _domain_id ts event =
-  let note = Play.event_to_note tones event in
+  let { Midi.Scale.note; Midi.Scale.volume } = Play.event_to_note tones event in
   match adjust_time ts with
   | None -> ()
   | Some ts ->
-      Midi.(
-        write_output device
-          [ message_on ~note ~timestamp:ts (* ~volume:'\070' *) () ]);
+      Midi.(write_output device [ message_off ~note ~timestamp:ts ~volume () ]);
       Printf.printf "%f: start of %s. ts: %ld\n%!" (Sys.time ())
         (Runtime_events.runtime_phase_name event)
         ts
 
 let runtime_end device tones _domain_id ts event =
-  let note = Play.event_to_note tones event in
+  let { Midi.Scale.note; Midi.Scale.volume } = Play.event_to_note tones event in
```

Comment:
I think here you can omit the Midi.Scale. in front of `volume`. It being stated for `note` is enough for OCaml to understand you are referring to a Midi.Scale type.

Reply:
Yes, I wrote this just to make it seem a little uniform! :D